custom_components/evse_load_balancer/meters/amsleser_meter.py

```diff
@@ -0,0 +1,121 @@
+"""AmsLeser implementation."""
+
+import logging
+from math import floor
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import (
+    DeviceEntry,
+)
+
+from .. import config_flow as cf  # noqa: TID252
+from ..ha_device import HaDevice  # noqa: TID252
+from .meter import Meter, Phase
+
+_LOGGER = logging.getLogger(__name__)
+
+# Mapping entities from the AmsLeser MQTT device based on suffixes.
+# Note that there is no support for production atm.
+# @see https://wiki.amsleser.no/en/api/mqtt
+ENTITY_REGISTRATION_MAP: dict[str, dict[str, str]] = {
+    cf.CONF_PHASE_KEY_ONE: {
+        # cf.CONF_PHASE_SENSOR_PRODUCTION: "PO1",
+        cf.CONF_PHASE_SENSOR_CONSUMPTION: "P1",
+        cf.CONF_PHASE_SENSOR_VOLTAGE: "U1",
+    },
+    cf.CONF_PHASE_KEY_TWO: {
+        # cf.CONF_PHASE_SENSOR_PRODUCTION: "PO2",
+        cf.CONF_PHASE_SENSOR_CONSUMPTION: "P2",
+        cf.CONF_PHASE_SENSOR_VOLTAGE: "U2",
+    },
+    cf.CONF_PHASE_KEY_THREE: {
+        # cf.CONF_PHASE_SENSOR_PRODUCTION: "PO3",
+        cf.CONF_PHASE_SENSOR_CONSUMPTION: "P3",
+        cf.CONF_PHASE_SENSOR_VOLTAGE: "U3",
+    },
+}
+
+
+class AmsleserMeter(Meter, HaDevice):
+    """AmsLeser implementation of the Meter class."""
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, device_entry: DeviceEntry
+    ) -> None:
+        """Initialize the Meter instance."""
+        Meter.__init__(self, hass, config_entry)
+        HaDevice.__init__(self, hass, device_entry)
+        self.refresh_entities()
+
+    def get_active_phase_current(self, phase: Phase) -> int | None:
+        """Return the active current on a given phase."""
+        active_power_w = self.get_active_phase_power(phase)
+        voltage_state = self._get_entity_state_for_phase_sensor(
+            phase, cf.CONF_PHASE_SENSOR_VOLTAGE
+        )
+
+        if None in [active_power_w, voltage_state]:
+            _LOGGER.warning(
+                (
+                    "Missing states for one of phase %s: active_power: %s, ",
+                    "voltage: %s. Are the entities enabled?",
+                ),
+                phase,
+                active_power_w,
+                voltage_state,
+            )
+            return None
+        # convert kW to W in order to calculate the current
+        return floor(active_power_w * 1000.0 / voltage_state) if voltage_state else None
```

Comment:
Little sanity check: according to [this post](https://community.home-assistant.io/t/evse-load-balancer-a-universal-load-balancer-for-your-ev-chargers/883863/49?u=dirkgroenen) the AMSleser sensors return a positive "power value" when delivering back to the grid, and a negative power value when consuming from the grid. 

Is this something you can confirm?  If that's the case we'll have to make sure to normalise the returned current value, as the load balancer expects a **positive current value** when importing and a **negative current value** when exporting.

Reply:
Huh, that's odd. I'm using the standard AmsLeser Pow-K meter with firmware 2.2.13 which is only one minor version behind the newest. I can see from the raw MQTT messages that it's sending over P1, P2 and P3 fields with the values I'm using in this PR.

I'll reach out to the manufacturer and ask if the P1, P2 and P3 fields are obsolete or just undocumented.

Reply:
I finally got around to implement the fallback, but I haven't tested it yet, and also haven't made new test cases yet. Hopefully I'll be able to do that in the coming week.

Reply:
@dirkgroenen The tests are in now :)